speck/src/lib.rs

```diff
@@ -67,6 +65,7 @@ pub fn encrypt_block(m: u128, k: u128) -> u128 {
 ///
 /// This precomputes a key schedule, which can then be used for both encrypting and decrypting
 /// messages.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
```

Comment:
`Debug` is probably a bad idea, as it would mean that you could accidentally leak a key. If anything, it should be implemented to write something like `****`.